tests/integration/buildcmd/test_build_samconfig.py

```diff
@@ -0,0 +1,133 @@
+from tests.testing_utils import run_command
+
+
+LOG = logging.getLogger(__name__)
+
+configs = {
+    ".toml": "samconfig/samconfig.toml",
+    ".yaml": "samconfig/samconfig.yaml",
+    ".json": "samconfig/samconfig.json",
+}
+
+
+class TestSamConfigWithBuild(BuildIntegBase):
+    @parameterized.expand(
+        [
+            (".toml"),
+            (".yaml"),
+            (".json"),
+        ]
+    )
+    def test_samconfig_works_with_extension(self, extension):
+        cmdlist = self.get_command_list(config_file=configs[extension])
+
+        LOG.info("Running Command: %s", cmdlist)
+        command_result = run_command(cmdlist, cwd=self.working_dir)
+        stdout = str(command_result[1])
+        stderr = str(command_result[2])
+
+        self.assertEqual(command_result.process.returncode, 0, "Build should succeed")
+        self.assertIn(
+            f"Built Artifacts  : {extension}",
+            stdout,
+            f"Build template should use build_dir from samconfig{extension}",
+        )
+        self.assertIn("Starting Build use cache", stderr, f"'cache'=true should be set in samconfig{extension}")
+
+    @parameterized.expand(
+        [
+            (".toml"),
+            (".yaml"),
+            (".json"),
+        ]
+    )
+    def test_samconfig_parameters_are_overridden(self, extension):
+        overrides = {"Runtime": "python3.8"}
+        overridden_build_dir = f"override_{extension}"
+
+        cmdlist = self.get_command_list(
+            config_file=configs[extension], parameter_overrides=overrides, build_dir=overridden_build_dir
+        )
+
+        LOG.info("Running Command: %s", cmdlist)
+        command_result = run_command(cmdlist, cwd=self.working_dir)
+        stdout = str(command_result[1])
+        stderr = str(command_result[2])
+
+        self.assertEqual(command_result.process.returncode, 0, "Build should succeed")
+        self.assertNotIn(
+            f"Built Artifacts  : {extension}",
+            stdout,
+            f"Build template should not use build_dir from samconfig{extension}",
+        )
+        self.assertIn(
+            f"Built Artifacts  : {overridden_build_dir}", stdout, f"Build template should use overridden build_dir"
+        )
+        self.assertIn("Starting Build use cache", stderr, f"'cache'=true should be set in samconfig{extension}")
+        self.assertNotIn("python3.9", stderr, f"parameter_overrides runtime should not read from samconfig{extension}")
+        self.assertIn(overrides["Runtime"], stderr, "parameter_overrides should use overridden runtime")
+        self.assertNotIn("SomeURI", stderr, f"parameter_overrides should not read ANY values from samconfig{extension}")
+
+
+@parameterized_class(
+    [  # Ordered by expected priority
+        {"extensions": [".toml", ".yaml", ".json"]},
+        {"extensions": [".yaml", ".json"]},
+    ]
+)
+class TestSamConfigExtensionHierarchy(BuildIntegBase):
+    testing_wd = Path(os.getcwd(), "tests", "integration")
```

Comment:
The `testing_wd` attribute isn't a directory for testing, but is the part of the codebase where the test data is stored. It's needed so that the various `samconfig` files can be copied over to a separate folder, to test that the correct one will be used if there are multiple in the same location. Perhaps I could rename to something more explicit of that functionality?

Reply:
Discussed offline to use the existing `test_data_path` attribute instead of creating a new one.